ql/src/java/org/apache/hadoop/hive/ql/parse/CalcitePlanner.java

```diff
@@ -3025,8 +3025,7 @@ private RelNode genTableLogicalPlan(String tableAlias, QB qb) throws SemanticExc
                 db, tabNameToTabObject, partitionCache, colStatsCache, noColsMissingStats);
             final TableScan scan = new HiveTableScan(cluster, cluster.traitSetOf(HiveRelNode.CONVENTION),
                 optTable, null == tableAlias ? tabMetaData.getTableName() : tableAlias,
-                getAliasId(tableAlias, qb), HiveConf.getBoolVar(conf,
-                    HiveConf.ConfVars.HIVE_CBO_RETPATH_HIVEOP), qb.isInsideView()
+                getAliasId(tableAlias, qb), ReturnPathManager.shouldUse(), qb.isInsideView()
```

Comment:
It may not be necessary to pass that boolean when we create HiveTableScan operators anymore, since there was some follow-up work that included the alias with the table name in the digest. If that is the case, we can remove it. Otherwise, it seems it is OK that we pass true even if finally we end up bailing out from return path when we check the operators/expressions in the plan.

Reply:
Ok, so in conclusion should this be set to true, hard coded?

Reply:
We should try with `false` hardcoded. If we set it to `true` by default, I believe it will change aliases for all plans once we enable return path (which also makes it more difficult to debug queries since these are synthetic aliases vs the aliases specified in the SQL query). Note that the boolean is used in `explainTerms` in `HiveTableScan` and the path for value `false` has changed since this boolean was created: I do not recall all details, but it may be that things work properly currently if we just set the value to `false`. A good way to verify would be to run all q tests matching `cbo_.*` hardcoding it to `false` in the `HiveTableScan` creation.

Reply:
hard coded false seems to work